circuit/src/verifier/transcript/mod.rs

```diff
@@ -136,29 +136,23 @@ where
 {
 	/// Read a scalar.
 	fn read_scalar(&mut self) -> Result<Halo2LScalar<C, L, P>, snark_verifier::Error> {
-		let mut data = <C::Scalar as PrimeField>::Repr::default();
-		self.reader.read_exact(data.as_mut()).map_err(|err| {
-			VerifierError::Transcript(
-				err.kind(),
-				"invalid field element encoding in proof".to_string(),
-			)
-		})?;
-
-		let scalar = Option::from(C::Scalar::from_repr(data)).ok_or_else(|| {
-			VerifierError::Transcript(
-				ErrorKind::Other,
-				"invalid field element encoding in proof".to_string(),
-			)
-		})?;
+		let scalar = self.reader.as_mut().and_then(|reader| {
+			let mut data = <C::Scalar as PrimeField>::Repr::default();
+			if reader.read_exact(data.as_mut()).is_err() {
+				return Value::unknown();
+			}
+			Option::<C::Scalar>::from(C::Scalar::from_repr(data))
+				.map(Value::known)
+				.unwrap_or_else(Value::unknown)
+		});
```

Comment:
We should leave error handling here as we did before.
If we fail to read from the reader, we shouldn't return `Value::unknown()`, instead we should return error.

Reply:
Because `Reader` type is wrapped in `Value` type, it doesn't allow me to return error. Instead I can manually print the error if it returns error inside and then return `Value::unknown()`. If I carry return inside the `Value` type it gives type errors when I try to assign them to a region.